web/admin/src/App.js

```diff
@@ -656,156 +496,183 @@ export default class App extends Component {
           </div>
           )
         }
+
         if (this.state.showAnswer === true){
           var block = false
           var header = false
           return(
           <div className="questionContainer">
-            {this.showButtons(questions)}
+            <CustomHeader
+            questions = {questions}
+            handleClick = {this.handleClick}
+            handleAnswer = {this.handleAnswer}
+            answerAll = {this.answerAll}
+            showBlock = {this.state.showBlock}
+            showAnswer = {this.state.showAnswer}
+            handleApproved = {this.handleApproved}
+            />
             {this.renderAnswered(questions, time)}
           </div>
           )
         }
         else {
           return(
           <div className="questionContainer">
-            {this.showButtons(questions)}
+            <CustomHeader
+            questions = {questions}
+            handleClick = {this.handleClick}
+            handleAnswer = {this.handleAnswer}
+            answerAll = {this.answerAll}
+            showBlock = {this.state.showBlock}
+            showAnswer = {this.state.showAnswer}
+            handleApproved = {this.handleApproved}
+            />
             {this.renderBlocked(questions, time)}
           </div>
           )
         }
       }
+    }
 
+    else {
+      return(
+        <div className="questionContainer">
+          <CustomHeader
+          questions = {questions}
+          handleClick = {this.handleClick}
+          handleAnswer = {this.handleAnswer}
+          answerAll = {this.answerAll}
+          showBlock = {this.state.showBlock}
+          showAnswer = {this.state.showAnswer}
+          handleApproved = {this.handleApproved}
+          />
+          {this.renderBlocked(questions, time)}
+        </div>
+      )
+    }
+  }
 
+  doDateMath = (date, time) => {
```

Comment:
Have you looked into [moment](http://momentjs.com/docs/#/displaying/fromnow/) to handle this?